media/admin/com_virtuemart/models/zasilkovna.php

```diff
@@ -388,106 +390,91 @@ public function raiseErrors()
-                if($first) {
-                    $q .= " (";
-                    $first = false;
-                }
-                else {
-                    $q .= ", (";
-                }
-                $streetEscaped = $db->escape($branch->nameStreet);
-                $q .= "'$branch->id', '$streetEscaped','$branch->currency','$branch->country')";
-
-            }
-            $db->setQuery($q);
-            $db->query();
-        }
-        else {
+        $xml = @simplexml_load_file($path);
+        if ($xml === false) {
             return false;
         }
 
-        return true;
-    }
-
-    /**
-     * @param $path
-     * @return bool
-     */
-    private function isFileUpToDate($path) {
-        if(!file_exists($path)) return false;
-        if(filemtime($path) < time() - (60 * 60 * 24)) return false;
-        if(filesize($path) <= 1024) return false;
+        $carrierIdsToDelete = $this->carrierRepository->getAllActiveCarrierIds();
+        foreach ($xml->carriers->carrier as $carrier) {
+            unset($carrierIdsToDelete[(string)$carrier->id]);
+
+            $data = [
+                'id' => $carrier->id,
+                'name' => $carrier->name,
+                'is_pickup_points' => $this->transformStringBoolToInt($carrier->pickupPoints),
+                'has_carrier_direct_label' => $this->transformStringBoolToInt($carrier->apiAllowed),
+                'separate_house_number' => $this->transformStringBoolToInt($carrier->separateHouseNumber),
+                'customs_declarations' => $this->transformStringBoolToInt($carrier->customsDeclarations),
+                'requires_email' => $this->transformStringBoolToInt($carrier->requiresEmail),
+                'requires_phone' => $this->transformStringBoolToInt($carrier->requiresPhone),
+                'requires_size' => $this->transformStringBoolToInt($carrier->requiresSize),
+                'disallows_cod' => $this->transformStringBoolToInt($carrier->disallowsCod),
+                'country' => $carrier->country,
+                'currency' => $carrier->currency,
+                'max_weight' => $carrier->maxWeight,
+                'deleted' => 0,
+            ];
+
+            if ($this->carrierRepository->carrierExists($carrier->id)) {
+                $this->carrierRepository->updateCarrier($carrier->id, $data);
+            } else {
+                $this->carrierRepository->insertCarrier($data);
+            }
+        }
 
+        $this->carrierRepository->setCarriersDeleted($carrierIdsToDelete);
         return true;
     }
 
     /**
-     * @param $path
-     * @return bool
+     * @param string $value
+     * @return int
      */
-    private function isFileUsable($path)//true if not older than 5 days
-    {
-        if(!file_exists($path)) return false;
-        if(filemtime($path) < time() - (60 * 60 * 24 * 5)) return false;
-        if(filesize($path) <= 1024) return false;
+    private function transformStringBoolToInt($value) {
```

Comment:
Není to zbytečně složité, když by stačilo ((string)$value === 'false' ? 0 : 1) ?

Reply:
to mě přijde nepřehledné

Reply:
Můžeme mít v metodě, ale proč mít podmínku, dvě přetypování a ternár místo jednoho ternáru s přetypováním?

Reply:
Jarda souhlasí s tím, že by se to mělo zjednodušit (ani přetypování na string není asi potřeba) a navrhuje, aby metoda vracela boolean, což by mělo být zohledněno i v CREATE a ALTER dotazech.

Reply:
upraveno. přetypování na string nutné je

Reply:
Proč ne return ((string)$value === 'false' ? false : true) ? Počítáš, že na vstupu bude třeba '0'?